samples/Dantooine/Dantooine.Client/Server/Startup.cs

```diff
@@ -0,0 +1,132 @@
+                options.DefaultChallengeScheme = OpenIdConnectDefaults.AuthenticationScheme;
+            })
+           .AddCookie()
+           .AddOpenIdConnect(options =>
+           {
+               options.SignInScheme = "Cookies";
+               options.Authority = openIDConnectSettings["Authority"];
+               options.ClientId = openIDConnectSettings["ClientId"];
+               options.ClientSecret = openIDConnectSettings["ClientSecret"];
+               options.RequireHttpsMetadata = true;
+               options.ResponseType = "code";
+               options.UsePkce = true;
+               options.Scope.Add("profile");
+               options.Scope.Add("api1");
+               options.SaveTokens = true;
+               options.GetClaimsFromUserInfoEndpoint = true;
+           });
+
+            services.AddControllersWithViews(options =>
+                 options.Filters.Add(new AutoValidateAntiforgeryTokenAttribute()));
+
+            services.AddRazorPages().AddMvcOptions(options =>
+            {
+                //var policy = new AuthorizationPolicyBuilder()
+                //    .RequireAuthenticatedUser()
+                //    .Build();
+                //options.Filters.Add(new AuthorizeFilter(policy));
+            });
+
+
+            services.AddHttpForwarder();
+        }
+
+        public void Configure(IApplicationBuilder app, IWebHostEnvironment env, IHttpForwarder forwarder)
+        {
+            if (env.IsDevelopment())
+            {
+                app.UseDeveloperExceptionPage();
+                app.UseWebAssemblyDebugging();
+            }
+            else
+            {
+                app.UseExceptionHandler("/Error");
+            }
+
+            app.UseSecurityHeaders(
+                SecurityHeadersDefinitions.GetHeaderPolicyCollection(env.IsDevelopment(),
+                    Configuration["OpenIDConnectSettings:Authority"]));
+
+            app.UseHttpsRedirection();
+            app.UseBlazorFrameworkFiles();
+            app.UseStaticFiles();
+
+            app.UseRouting();
+            app.UseAuthentication();
+            app.UseAuthorization();
+
+            var httpClient = new HttpMessageInvoker(new SocketsHttpHandler()
+            {
+                UseProxy = false,
+                AllowAutoRedirect = false,
+                AutomaticDecompression = DecompressionMethods.None,
+                UseCookies = false
+            });
+            var transformer = new CookieTokenTransformer(); // or HttpTransformer.Default;
+            var requestConfig = new ForwarderRequestConfig { ActivityTimeout = TimeSpan.FromSeconds(100) };
+
+            app.UseEndpoints(endpoints =>
+            {
+                endpoints.MapRazorPages();
+                endpoints.MapControllers();
+                endpoints.Map("/api/DantooineApi1", async httpContext =>
+                {
+                    var error = await forwarder.SendAsync(httpContext, "https://localhost:44342/",
```

Comment:
I'm not familiar with YARP but is it something we have to do ourselves? It seems weird to me we have to allocate the HTTP client and the transformer and manually do the forwarding stuff here.

Reply:
There is many ways of doing this. I though I would just use the most simple. The code is just a demo. You would have to use a token handle anyway to validate if the token is not expired, maybe update the token or add caching. Lots of stuff you can do here and each roll requires something different.

Reply:
improved with general implementation